test/distributed/_shard/sharded_tensor/test_sharded_tensor.py

```diff
@@ -1464,6 +1465,61 @@ def test_gather_uneven(self) -> None:
         else:
             self.assertIsNone(full_tensor)
 
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_to_cpu(self):
+        cpu_spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cpu",
+                "rank:1/cpu",
+                "rank:2/cpu",
+                "rank:3/cpu",
+            ],
+        )
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
```

Comment:
Maybe we should also test a spec where some shards are on cpu and some on GPU and then we call `.cpu()`.

Reply:
hmmm how we should represent such a ShardedTensor, is it being managed by one process group? Maybe a processgroupGloo can manage a ShardedTensor where local shards can be on CPU/GPU together?

Reply:
added.